features/graph/api/src/main/java/org/opennms/netmgt/graph/api/generic/GenericGraph.java

```diff
@@ -119,10 +121,16 @@ public Focus getDefaultFocus() {
         return defaultFocus;
     }
 
-    //    @Override
-//    public Vertex getVertex(NodeRef nodeRef) {
-//        return nodeRefToVertexMap.get(nodeRef);
-//    }
+    @Override
+    public GenericVertex getVertex(NodeRef nodeRef) {
+        Objects.requireNonNull(nodeRef);
+        for (NodeRef eachVariant : nodeRef.getVariants()) {
+            if (nodeRefToVertexMap.containsKey(eachVariant)) {
+                return nodeRefToVertexMap.get(eachVariant);
+            }
+        }
+        return null;
```

Comment:
A general thought here: since we are designing a new module: 
How is our stance towards Optional as return parameter vs. return null? I think that is the preferred way these days to handle possible null return values?

Reply:
In most cases Optionals are just not necessary. In the rare cases where an Optional would be neat, in my opinion, don't make up for the need to use them everywhere. Especially on an API level. The reason for this is, that you have to document the API level anyways and "must not be null" or "may be null" is as sufficient as the optional. And if it really helps, you can always create an Optional and use it.

TL;DR: may be nice, but not necessary. If you need it for exception handling or so, just create it ad-hoc, instead of using it everywhere.